astropy/wcs/wcs.py

```diff
@@ -129,6 +130,8 @@
 # Additional relax bit flags
 WCSHDO_SIP = 0x10000
 
+# Regular expression defining SIP keyword
+SIP_KW = re.compile('''^(A|B|AP|BP)_(?:[0-9]|1[0-9])_(?:[0-9]|1[0-9])(|[A-Z])$''')
```

Comment:
That regex matches all keyword combinations from that nested loop that I removed. 
I have no idea what is the source of it, as it doesn't follow the original SIP specification [1]
Following git blame @mdboom do you have any insight on it?

[1] http://fits.gsfc.nasa.gov/registry/sip/SIP_distortion_v1_0.pdf